### PR TITLE
for #11698 added on demand workmanager initialization

### PR DESCRIPTION
**Pull request by sraturi**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -281,6 +281,10 @@
         <meta-data
             android:name="firebase_analytics_collection_enabled"
             android:value="false" />
+        <provider
```
Comment (NotWoods) [inline app/src/main/AndroidManifest.xml:284]: Might be worthwhile to add a comment here, its not immediately clear at a glance that this removes the work manager initializer (even though this is what the docs recommend)

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 2]
```diff
@@ -54,7 +54,7 @@ import org.mozilla.fenix.utils.Settings
  *  Installs [CrashReporter], initializes [Glean]  in fenix builds and setup Megazord in the main process.
  */
 @Suppress("Registered", "TooManyFunctions", "LargeClass")
-open class FenixApplication : LocaleAwareApplication() {
+open class FenixApplication : LocaleAwareApplication(), androidx.work.Configuration.Provider {
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/FenixApplication.kt:57]:
```suggestion
open class FenixApplication : LocaleAwareApplication(), Configuration.Provider {
```

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 3]
```diff
@@ -404,4 +404,9 @@ open class FenixApplication : LocaleAwareApplication() {
     companion object {
         private const val KINTO_ENDPOINT_PROD = "https://firefox.settings.services.mozilla.com/v1"
     }
+
+    override fun getWorkManagerConfiguration() =
+        androidx.work.Configuration.Builder()
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/FenixApplication.kt:409]:
```suggestion
        Configuration.Builder()
```

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (sraturi) [reply app/src/main/AndroidManifest.xml:284]: yh for sure. @mcomella  quickly wanted to see the changes for #11698  (did some performance test for the workmanager). Im not sure if we are landing this yet.

**Comment by csadilek** [comment]
@sraturi @mcomella please, let's not land this before more folks had a chance to look at it, and after extensive testing. Need to make sure sync, glean, add-ons workers are all still working as expected.

**Comment by Amejia481** [comment]
I tested the add-on flows that usages `workmanager` and they are working as expected ✅ 
**Please, do not merge yet, as we need to check for sync and glean.**

**Pull request merged**